app/code/Magento/Theme/Test/Unit/Block/Html/Header/LogoTest.php

```diff
@@ -89,4 +89,34 @@ public function testGetLogoWidth()
 
         $this->assertEquals('170', $block->getLogoHeight());
     }
+
+    /**
+     * @covers \Magento\Theme\Block\Html\Header\Logo::getLogoImageWidth
+     */
+    public function testGetLogoImageWidth()
+    {
+        $scopeConfigWithLogoWidthArgument = $this->getMockForAbstractClass(ScopeConfigInterface::class);
+        $scopeConfigWithLogoWidthArgument->expects($this->any())->method('getValue')->willReturn(null);
+
+        $scopeConfigWithoutLogoWidthArgument = $this->getMockForAbstractClass(ScopeConfigInterface::class);
+        $scopeConfigWithoutLogoWidthArgument->expects($this->once())->method('getValue')->willReturn(800);
+
+        $objectManager = new ObjectManager($this);
+        $argumentsWithLogoWidth = [
+            'scopeConfig' => $scopeConfigWithLogoWidthArgument,
+            '_data' => [
+                'logo_width' => 200
+            ]
+        ];
+        $blockWithLogoWidthArgument = $objectManager->getObject(Logo::class, $argumentsWithLogoWidth);
+
+        $this->assertEquals('200', $blockWithLogoWidthArgument->getLogoWidth());
+
+        $argumentsWithoutLogoWidth = [
+            'scopeConfig' => $scopeConfigWithoutLogoWidthArgument
+        ];
+        $blockWithoutLogoWidthArgument = $objectManager->getObject(Logo::class, $argumentsWithoutLogoWidth);
+
+        $this->assertEquals('800', $blockWithoutLogoWidthArgument->getLogoWidth());
+    }
```

Comment:
Could you split that test into 2:
- `testGetLogoImageWidthWhenImageWidthConfigured`
- `testGetLogoImageWidthWhenImageWidthNotConfigured`

That gives us 2 benefits:
- When it fails - we exactly know which case fails
- The test is easier to read and understand.

You can additionally split that into sections:
- `// Given` - The environmental / mocked elements (eg. ScopeConfig)
- `// When` - The call to tested method
- `// Then` - Assertions (comparison of expected to actual results)